### ISSUE TITLE
IE ComObject works unexpected 

### ISSUE THREAD

**Issue by tanaliko** [PowerShell/PowerShell#12146]
$ie.Document.getElementsByTagName("input") 
produces error 
MethodException: Cannot find an overload for "getElementsByTagName" and the argument count: "1". 

```
PS C:\> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.0.0
PSEdition                      Core
GitCommitId                    7.0.0
OS                             Microsoft Windows 10.0.18363
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

PS C:\> $ie = new-object -ComObject "InternetExplorer.Application"
PS C:\> $ie.Visible=$true
PS C:\> while($ie.Busy) { Start-Sleep -Milliseconds 100 }
PS C:\> $ie.Navigate("https://www.microsoft.com/")
PS C:\> while($ie.Busy) { Start-Sleep -Milliseconds 100 }
PS C:\> $ie.Document.getElementsByTagName("input")
MethodException: Cannot find an overload for "getElementsByTagName" and the argument count: "1".
PS C:\>
```

# Steps to reproduce
Test script  to run
```
$PSVersionTable 
$ie = new-object -ComObject "InternetExplorer.Application"
$ie.Visible=$true
while($ie.Busy) { Start-Sleep -Milliseconds 100 }
$ie.Navigate("https://www.microsoft.com/")
while($ie.Busy) { Start-Sleep -Milliseconds 100 }
$ie.Document.getElementsByTagName("input") 
```

# Expected behavior
as result we have an object that we can operate 

# Actual behavior
$ie.Document.getElementsByTagName("input") produces error MethodException: Cannot find an overload for "getElementsByTagName" and the argument count: "1"

#Addtional info 
On the same mashine system powershell works as expected 
```
PS C:\> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      5.1.18362.628
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.18362.628
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1


PS C:\> $ie = new-object -ComObject "InternetExplorer.Application"
PS C:\> $ie.Visible=$true
PS C:\> while($ie.Busy) { Start-Sleep -Milliseconds 100 }
PS C:\> $ie.Navigate("https://www.microsoft.com/")
PS C:\> while($ie.Busy) { Start-Sleep -Milliseconds 100 }
PS C:\> $ie.Document.getElementsByTagName("input") | ft

className id                         tagName parentElement      style              onhelp onclick ondblclick onkeydown
--------- --                         ------- -------------      -----              ------ ------- ---------- ---------
          cli_shellHeaderSearchInput INPUT   System.__ComObject System.__ComObject
PS C:\>
```

`ft` added since a lot of output

and big difference detected on `$ie.Document | Get-Member | Measure-Object` output 
in  PowerShell  7.0 - **67** methods (no events, no properties) `TypeName: System.__ComObject#{c59c6b12-f6c1-11cf-8835-00a0c911e8b2}`
in system PowerShell 5.1 - **527** methods, events and propertiess `TypeName: mshtml.HTMLDocumentClass`

**Comment by vexx32** [comment]
Check the overload definitions for the method by calling:
```
$ie.Document.getElementsByTagName
```

But yes, in general COM objects are more restricted in .NET Core as I understand it. If you need that functionality, the .NET Core team would have to implement that functionality, I think.

**Comment by tanaliko** [comment]
7.0
'''
PS C:\> $ie.Document.getElementsByTagName
OverloadDefinitions
-------------------
Variant getElementsByTagName () 
'''
5.1 
'''
PS C:\> $ie.Document.getElementsByTagName

OverloadDefinitions
-------------------
mshtml.IHTMLElementCollection, Microsoft.mshtml, Version=7.0.3300.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a g
etElementsByTagName(string v)
mshtml.IHTMLElementCollection, Microsoft.mshtml, Version=7.0.3300.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a D
ispHTMLDocument.getElementsByTagName(string v)
mshtml.IHTMLElementCollection, Microsoft.mshtml, Version=7.0.3300.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a I
HTMLDocument3.getElementsByTagName(string v)
'''

**Comment by topfunet** [comment]
I also have this problem, can anyone help?

**Comment by vexx32** [comment]
/cc @daxian-dbw this looks like it might be an issue with the COM binder perhaps?

@topfunet @tanaliko can you two verify this is still happening on the latest release version of pwsh?

**Comment by topfunet** [comment]
I'm using PowerShell 7.0.3, and this problem is still exist.

**Comment by doctordns** [comment]
I am testing in 7.2 P1 and this code works fine.

**Comment by vexx32** [comment]
@topfunet please check with 7.1 or one of the more recent 7.2 preview releases?

**Comment by topfunet** [comment]
I'm new to PowerShell, and I have 2 questions:
1. Does PowerShell 7.1 support windows 7 sp1?
2. Does Microsoft.PowerShell.SDK 7.1 support .net core 3.1?

Because according to this link
https://docs.microsoft.com/en-us/powershell/scripting/whats-new/what-s-new-in-powershell-71?view=powershell-7.1
PowerShell7.0 seems to build for .net5.0 and only can run on Windows 8.1/10.

**Comment by vexx32** [comment]
I don't know of any specific reasons .net 5.0 _wouldn't_ work on win7 sp1 to be honest, but it's possible there may be some issues there.

I don't know whether the COM object fixes can be backported to 7.0.x though, from memory there were changes to the COM API in .NET itself which might mean backporting them to make things work properly in 7.0.x is not possible.

**Comment by topfunet** [comment]
Seem like PowerShell 5.1 work perfectly.
But I just tried, Microsoft.PowerShell.5.1.ReferenceAssemblies won't work in .net core 3.1.
Any suggestion?

**Comment by iSazonov** [comment]
I guess this was fixed in #13304
